src/components/Auth/Login.tsx

```diff
@@ -2,24 +2,33 @@ import { useForm } from 'react-hook-form';
 import { Link } from 'react-router-dom';
 import { BsFacebook } from 'react-icons/bs';
 import { FcGoogle } from 'react-icons/fc';
+import { signInWithEmailAndPassword } from 'firebase/auth';
+import { useState } from 'react';
 import AuthImages from '../helpers/AuthImages';
-import { LoginData } from '../../types';
+import { LoginData, LoginErrors } from '../../types';
+import { logInWithFacebook, logInWithGoogle } from './loginProviders';
+import { auth } from '../configuration/firebase';
+import { UseUpdateDoc } from '../hooks/useManageDoc';
+import loginErrors from '../helpers/loginErrors';
 
-interface LoginProps {
-  logInWithFacebook: () => void;
-  logInWithGoogle: () => void;
-  loginError: string;
-  loginUser: (user: LoginData) => void;
-}
-
-function Login({ logInWithFacebook, logInWithGoogle, loginError, loginUser }: LoginProps) {
+function Login() {
+  const [loginError, setLoginError] = useState('');
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<LoginData>();
 
-  const onSubmit = (data: LoginData) => loginUser(data);
+  const onSubmit = (data: LoginData) =>
+    signInWithEmailAndPassword(auth, data.email, data.password)
+      .then(async (userCredential) => {
+        const { user } = userCredential;
+        await UseUpdateDoc('users', [user.uid], { isOnline: true });
+      })
+      .catch(({ code }: { code: keyof LoginErrors }) => {
```

Comment:
Nie byłoby lepiej wstawić tutaj cały interfejs?
`.catch(( { code }: interfejsObiektuBłędu ) => {`
Czy może koniecznie musisz wskazać, że `code` to`keyof LoginErrors` ? Bo jeśli tak, to może problem leży w nadrzędnym interfejsie, którego częścią jest klucz `code`?

Reply:
Użyłem keyof ponieważ code to jest klucz obiektu a dopiero pozniej zwracam wartosc chociaż moze masz racje ze jest coś zle przemyślane
![image](https://user-images.githubusercontent.com/26391807/167105399-51adf48a-eca6-4562-bcfd-887b8eb80fbe.png)

Reply:
Bardziej chodzi mi o to, że używasz interfejsu tylko dla jednego klucza w zwracanym obiekcie, gdy raczej powinieneś dla całego obiektu, bo też i cały obiekt dostajesz. I to keyof to powinien być tylko fragment tego interfejsu. Przyjdzie potrzeba użycia innego fragmentu z obiektu błędu i całość się zacznie sypać.

Reply:
Wydaje mi się ze tylko jeden błąd wywala